packages/gatsby/src/state-machines/actions.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  assign,
+  AnyEventObject,
+  ActionFunction,
+  spawn,
+  ActionFunctionMap,
+  DoneEventObject,
+} from "xstate"
+import { Store } from "redux"
+import { IBuildContext, IMutationAction } from "../services"
+import { actions } from "../redux/actions"
+import { listenForMutations } from "../services/listen-for-mutations"
+import { DataLayerResult } from "./data-layer"
+import { assertStore } from "../utils/assert-store"
+
+export const callRealApi = async (
+  event: IMutationAction,
+  store?: Store
+): Promise<unknown> => {
+  assertStore(store)
```

Comment:
If we're asserting it, should it be optional in the params? I assume this is a TS assertion which makes TS accept that you access `store` unguarded below?

Reply:
Yes, it's optional as with all of the context values, and assertStore is both a runtime null check but also a TypeScript assert that narrows the type.